### PR TITLE
feat: Label metrics from live run with "live_run_id"

### LINKED ISSUE #1940: After Start/Stop in Dozer Live data keeps accumulating in the progress
If I start / stop / start dozer live the counts in the progress bar keeps increasing.
After a stop and a restart it should restart from 0

### PR DESCRIPTION
**Pull request by chubei**
Closes #1940 

Also the backend part of https://github.com/getdozer/dozer/issues/1915

### CODE REVIEW

**Comment by Jesse-Bakker** [comment]
Can we make a `metrics` `Recorder` that drives the progress bars instead of having to update the progress bars from components like `cache_builder`?

**Comment by chubei** [comment]
I don't want to keep track of a `HashMap<String, ProgressBar>` because that'll requrie interior mutability. Any way around this?

**Comment by chubei** [comment]
Oh I misunderstood. You meant to drive progress bar from metrics. I believe that's out of the scope of this PR.

File: dozer-cli/src/live/progress.rs [hunk 1]
```diff
@@ -61,3 +64,19 @@ pub async fn progress_stream(
         retry_interval.tick().await;
     }
 }
+
+fn labels_match(
+    prom_labels: &prometheus_parse::Labels,
+    dozer_labels: &dozer_tracing::Labels,
+) -> bool {
+    for (key, value) in dozer_labels.iter() {
```
Comment (Jesse-Bakker) [inline dozer-cli/src/live/progress.rs:72]:
```rust
dozer_labels.iter().all(|(key, value)| prom_labels.get(key) == Some(value))
```

**Review by Jesse-Bakker — APPROVED** [review]
Small style nit, otherwise LGTM!

File: dozer-cli/src/live/progress.rs (hunk 1 above)
Comment (chubei) [reply dozer-cli/src/live/progress.rs]: Much nicer!

**Comment by snork-alt** [comment]
@chubei Why does this require doc update ?

**Comment by chubei** [comment]
Sorry bogus auto labelling. Removing now.

**Pull request merged**